src/pages/[spaceId]/[...topic].tsx

```diff
@@ -32,27 +33,47 @@ export const getStaticPaths: GetStaticPaths = async () => {
   }
 }
 
-export const getStaticProps = getCommonStaticProps<{
-  dehydratedState: any
-  postId: string
-  title: string | null
-}>(
-  (data) => ({ head: { disableZoom: true, title: data.title } }),
+export const getStaticProps = getCommonStaticProps<
+  {
+    dehydratedState: any
+    title: string | null
+    desc: string | null
+  } & ChatPageProps
+>(
+  (data) => ({
+    head: { disableZoom: true, title: data.title, description: data.desc },
+  }),
   async (context) => {
-    const topic = context.params?.topic as string
-    const postId = getIdFromSlug(topic)
-    if (!postId) return undefined
+    const topicParams = context.params?.topic as string[]
+    if (topicParams.length <= 0 || topicParams.length > 2) {
+      return undefined
+    }
+
+    const [topic, chatId] = topicParams
+    const roomId = getIdFromSlug(topic)
+    if (!roomId) return undefined
+
+    const invalidChatId = chatId && isNaN(parseInt(chatId))
```

Comment:
I think the code for getting topic you can extract to separate function

Reply:
Updated